brainscore_vision/metrics/threshold/metric.py

```diff
@@ -0,0 +1,514 @@
+        if isinstance(source, float):
+            source_threshold = source
+        elif isinstance(source, BehavioralAssembly):
+            source_threshold = self.compute_threshold(source, self._independent_variable)
+            # check whether the psychometric function fit was successful - if not, return a score of 0
+            if source_threshold == 'fit_fail':
+                score = Score([0.], coords={'aggregation': ['center', ]}, dims=['aggregation'])
+                score.attrs['error'] = 0.
+                return score
+        else:
+            raise TypeError(f'source is type {type(source)}, but type BehavioralAssembly or float is required.')
+        return self.scoring_function(source_threshold, target)
+
+    def ceiling(self, assembly: Union[PropertyAssembly, Dict[str, PropertyAssembly]]) -> Score:
+        """
+        Computed by one-vs all for each of the NUM_TRIALS human indexes. One index is removed, and scored against
+        a pool of the other values.
+
+        Currently copied with modification from 'https://github.com/brain-score/brain-score/blob/
+        jacob2020_occlusion_depth_ordering/brainscore/metrics/data_cloud_comparision.py#L54'.
+
+        :param assembly: the human PropertyAssembly containing human responses, or a dict containing the
+                          PropertyAssemblies of the ThresholdElevation metric.
+        :return: Score object with coords center (ceiling) and error (STD)
+        """
+        human_thresholds: list = assembly.values.tolist()
+        scores = []
+        for i in range(len(human_thresholds)):
+            random_state = np.random.RandomState(i)
+            random_human_score = random_state.choice(human_thresholds, replace=False)
+            metric = Threshold(self._independent_variable, self.fit_function, self.fit_inverse_function,
+                               self.threshold_accuracy)
+            human_thresholds.remove(random_human_score)
+            score = metric(random_human_score, human_thresholds)
+            score = float(score[(score['aggregation'] == 'center')].values)
+            human_thresholds.append(random_human_score)
+            scores.append(score)
+
+        ceiling, ceiling_error = np.mean(scores), np.std(scores)
+        ceiling = Score([ceiling], coords={'aggregation': ['center']}, dims=['aggregation'])
+        ceiling.attrs['error'] = ceiling_error
+        return ceiling
+
+    def compute_threshold(self, source: BehavioralAssembly, independent_variable: str) -> Union[float, str]:
+        """Converts the source BehavioralAssembly to a threshold float value."""
+        assert len(source.values) == len(source[independent_variable].values)
+
+        x_points = source[independent_variable].values
+        accuracies = self.convert_proba_to_correct(source)
+        if np.mean(accuracies) < self.required_accuracy:
+            print('Psychometric threshold fit failure due to low accuracy.')
```

Comment:
recommend `logging` instead of `print` for all of these
```
import logging

_logger = logging.getLogger(__name__)

...

def myfunc():
    _logger.debug('Psychometric threshold ...')